app/scenes/Settings/Details.tsx

```diff
@@ -276,6 +282,31 @@ function Details() {
             />
           </SettingRow>
 
+          <SettingRow
+            border={false}
+            label={t("Table of Contents Position")}
+            name="tocPosition"
+            description={t(
+              "The side to display the table of contents in relation to the main content."
+            )}
+          >
+            <InputSelect
+              ariaLabel={t("TOC position")}
+              options={[
+                {
+                  label: "Left",
+                  value: TOCPosition.Left,
+                },
+                {
+                  label: "Right",
+                  value: TOCPosition.Right,
+                },
+              ]}
+              value={tocPosition}
+              onChange={(p: TOCPosition) => setTocPosition(p)}
```

Comment:
```suggestion
              onChange={setTocPosition}
```

Should work, no? Or is typescript forcing the intermediate function

Reply:
Yeah, Typescript 😅 .